src/utils/array.js

```diff
@@ -14,6 +14,34 @@
  * limitations under the License.
  */
 
+/**
+  * Compares if two arrays contains exactly same elements (of same number)
+  * It will NOT step into array elements if ever seen, simply compare current
+  * level references.
+  * Notice it does NOT handle NaN case as expected.
+  *
+  * @param {!Array<T>} arr1
+  * @param {!Array<T>} arr2
+  * @return {boolean}
+  * @template T
+  */
+export function areEqual(arr1, arr2) {
```

Comment:
@jridgewell What would be the most efficient way to do this? (just shallow equality)

Reply:
I can remove `.splice` by putting a dummy object at `index` if `splice` turns out to be of very bad performance. Restricted from using `Set`, I am not sure implementing something similar with a JS level hash would be faster than what we currently have, considering `options` often not too long

Reply:
depending on whether `order` is part of equality or not, this can be simplified. Most cases we would want order to be considered IMO. (specially in the use-case this is used).

I suggest:
rename to `areEqualOrdered(..)` to make that clear, then it should be doable in O(n) with a single loop, no copying.

Reply:
Chances are that the option elements might be reordered or moved into newly created child <div>s (dunno if current validation allows for changing hierarchy), while the elements themselves are intact. In that case, I believe the `order` may be different

Reply:
`amp-selector` needs to reinitialize if order changes (it keeps track of things such as `focusedIndex_` that are tied to the index of elements in the array) so we do want to do `areEqualOrdered` when deciding whether we need to reinitialize or not.